File: bestia-zoneserver/src/test/java/net/bestia/zoneserver/guild/GuildServiceTest.java
package net.bestia.zoneserver.guild;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import net.bestia.model.dao.GuildDAO;
import net.bestia.model.dao.GuildMemberDAO;
import net.bestia.model.dao.PlayerBestiaDAO;

@RunWith(MockitoJUnitRunner.class)
public class GuildServiceTest {

	private GuildService gs;
	
	@Mock
	private GuildDAO guildDao;
	
	@Mock
	private GuildMemberDAO guildMemberDao;
	
	@Mock
	private PlayerBestiaDAO bestiaDao;
	
	@Before
	public void setup() {
		gs = new GuildService(guildDao, guildMemberDao, bestiaDao);
	}

	public void addPlayerToGuild_existingGuildEnoughSpace_memberAdded() {
		
	}

	public void addPlayerToGuild_existingGuildFull_memberAdded() {

	}

	public void addPlayerToGuild_playerHasGuild_memberNotAdded() {

	}

	public void getMaxGuildMembers_notExistingGuild_0() {

	}

	public void getMaxGuildMembers_existingGuild_correctNumber() {

	}

	public void hasGuild_memberWithGuild_true() {

	}

	public void hasGuild_memberWithGuild_false() {

	}

	public void getGuildOfPlayer_memberHasNoGuild_empty() {

	}

	public void addExpTaxToGuild_notExistingPlayer_nothing() {

	}

	public void addExpTaxToGuild_negativeExp_nothing() {

	}

	public void addExpTaxToGuild_validPlayerAndExp_taxAdded() {

	}
	
	public void getNeededNextLevelExp_existingGuild_exp() {
		
	}
}

File: bestia-zoneserver/src/test/java/net/bestia/zoneserver/guild/GuildServiceTest.kt
package net.bestia.zoneserver.guild

import org.junit.Before
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner

import net.bestia.model.dao.GuildDAO
import net.bestia.model.dao.GuildMemberDAO
import net.bestia.model.dao.PlayerBestiaDAO

@RunWith(MockitoJUnitRunner::class)
class GuildServiceTest {

  private lateinit var gs: GuildService

  @Mock
  private lateinit var guildDao: GuildDAO

  @Mock
  private lateinit var guildMemberDao: GuildMemberDAO

  @Mock
  private lateinit var bestiaDao: PlayerBestiaDAO

  @Before
  fun setup() {
    gs = GuildService(guildDao, guildMemberDao, bestiaDao)
  }

  fun addPlayerToGuild_existingGuildEnoughSpace_memberAdded() {

  }

  fun addPlayerToGuild_existingGuildFull_memberAdded() {

  }

  fun addPlayerToGuild_playerHasGuild_memberNotAdded() {

  }

  fun getMaxGuildMembers_notExistingGuild_0() {

  }

  fun getMaxGuildMembers_existingGuild_correctNumber() {

  }

  fun hasGuild_memberWithGuild_true() {

  }

  fun hasGuild_memberWithGuild_false() {

  }

  fun getGuildOfPlayer_memberHasNoGuild_empty() {

  }

  fun addExpTaxToGuild_notExistingPlayer_nothing() {

  }

  fun addExpTaxToGuild_negativeExp_nothing() {

  }

  fun addExpTaxToGuild_validPlayerAndExp_taxAdded() {

  }

  fun getNeededNextLevelExp_existingGuild_exp() {

  }
}
